File: FreeSql.Generator.Core/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FreeSql.DataAnnotations;
using FreeSql.Generator.Core.WordsConvert;

namespace FreeSql.Generator.Core
{
    [Table(Name ="Project")]
    public class Project : IOutPut, IKey<long>
    {
        [Column(IsPrimary = true, IsIdentity = true)]
        public long Id { get; set; }
        /// <summary>
        /// 项目名——命名空间
        /// </summary>
        public string ProjectName { get; set; }
        /// <summary>
        /// 根目录
        /// </summary>
        public string RootPath { get; set; }
        /// <summary>
        /// 输出路径 默认Output
        /// </summary>
        public string OutPutPath { get; set; } = "Output";
        /// <summary>
        /// 生成器模式 CodeFirst根据实体代码来生成 DbFirst根据数据库生成代码
        /// </summary>
        public GeneratorMode GeneratorMode { get; set; } = GeneratorMode.CodeFirst;
        /// <summary>
        /// 实体基类名称 CodeFirst根据此项反射根据此项
        /// </summary>
        public string EntityBaseName { get; set; } = "EntityBase`1";
        /// <summary>
        /// 数据库信息
        /// </summary>
        public DataSource DataSource { get; set; }
        /// <summary>
        /// 数据库id
        /// </summary>
        public long DataSourceId { get; set; }
        /// <summary>
        /// 作者
        /// </summary>
        public string Author { get; set; }
        /// <summary>
        /// 数据表选项
        /// </summary>
        public Entity Entity { get; set; }
        public string IncludeTables { get; set; }
        public string IgnoreTables { get; set; }
        /// <summary>
        /// 只生成某些表
        /// </summary>
        [Column(IsIgnore = true)]
        public List<string> IncludeTable => IncludeTables?.Split(",").ToList();
        /// <summary>
        /// 忽略某些表
        /// </summary>
        [Column(IsIgnore = true)]
        public List<string> IgnoreTable => IgnoreTables?.Split(",").ToList();
        /// <summary>
        /// 多表构建器
        /// </summary>
        public List<BuilderOptions> Builders { get; set; } = new List<BuilderOptions>();
        /// <summary>
        /// 全表构建器
        /// </summary>
        public List<BuilderOptions> GlobalBuilders { get; set; } = new List<BuilderOptions>();

    }

    public interface IKey<out TKey>
    {
        TKey Id { get; }
    }
}

File: src/FreeSqlBuilderUI/Client/FreeSqlBuilderClient/ClientApp2/fsbuilderv2/src/app/core/services/interface/project.ts
export class Project {
  id: number;
  projectInfoId: number;
  projectInfo: ProjectInfo;
  generatorModeConfigId: number;
  generatorModeConfig: GeneratorModeConfig;
  builders: BuilderOptions[];
  buildersId: number[];
  _buildersId: number[];
  globalBuilders: BuilderOptions[];
  globalBuildersId: number[];
  _globalBuildersId: number[];
}

export class ProjectInfo {
  id: number;
  projectName: string;
  rootPath: string;
  outPutPath: string;
  author: string;
}

export class GeneratorModeConfig {
  id = 0;
  name = ``;
  generatorMode: GeneratorMode = GeneratorMode.DbFirst;
  dataSourceId = 0;
  dataSource: DataSource = new DataSource();
  entitySourceId = 0;
  entitySource: EntitySource = new EntitySource();
  pickType: PickType = PickType.Ignore;
  projects: Project[];
  includeTables = ``;
  ignoreTables = ``;
}

export enum PickType {
  Pick,
  Ignore,
}
export enum GeneratorMode {
  DbFirst,
  CodeFirst,
}
export class EntitySource {
  id: number;
  name = '';
  entityAssemblyName = '';
  entityBaseName = '';
}

export class DataSource {
  id = 0;
  name = '';
  dbType: DataType = DataType.SqlServer;
  connectionString = '';
  generatorModeConfigs: GeneratorModeConfig[];
}

export enum DataType {
  MySql,
  SqlServer,
  PostgreSQL,
  Oracle,
  Sqlite,
  OdbcOracle,
  OdbcSqlServer,
  OdbcMySql,
  OdbcPostgreSQL,
  Odbc,
  OdbcDameng,
  MsAccess,
  Dameng,
  OdbcKingbaseES,
  ShenTong,
}

export class BuilderOptions {
  id: number;
  strExtensionOptions: string;
  name: string;
  prefix: string;
  outPutPath: string;
  mode: ConvertMode;
  template: Template;
  templateId: number;
  suffix: string;
  type: BuilderType;
  fileExtensions: string;
  projects: Project[];
}

export enum BuilderType {
  Builder,
  GlobalBuilder,
}

export enum ConvertMode {
  None,
  AllLower,
  AllUpper,
  FirstUpper,
}

export class Template {
  id: number;
  templateName: string;
  templatePath: string;
  templateContent: string;
  fileExtension: string;
}
